### PR TITLE
SCHED-71: Initial mini-model implementation.

### PR DESCRIPTION
**Pull request by sraaphorst**
Initial implementation of the mini-model, minus output and some internal data structures. (Will add these tomorrow.)

### CODE REVIEW

File: common/structures/conditions.py [hunk 1]
```diff
@@ -196,3 +196,4 @@ def __init__(self,
                  wind: WindConditions):
         self.sky = sky
         self.wind = wind
+
```
Comment (sraaphorst) [inline common/structures/conditions.py:199]: I have no idea why it is saying there is a diff between these two files.

File: common/minimodel/__init__.py [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+    The wavelength must be specified in nanometers.
+    """
+    id: str
+    exec_time: timedelta
+    prog_time: timedelta
+    part_time: timedelta
+    observed: bool
+    qa_state: QAState
+    guide_state: bool
+    required_resources: Set[Resource]
```
Comment (stroncod) [inline common/minimodel/__init__.py:290]: Won't each atom have the same Resource in a sequence?

File: common/minimodel/__init__.py [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+    tag: TargetTag
+
+    def __post_init__(self):
+        super().__post_init__()
+
+
+@dataclass
+class Resource:
+    id: str
+    name: str
```
Comment (stroncod) [inline common/minimodel/__init__.py:267]: If I recall the Selector would need more information (FPU, Gratings, etc). Is this going to be separate from the mini model ?

**Review by stroncod — APPROVED** [review]
I'm not sure if you want this to be merged or just reviewed but I approved just in case

File: common/minimodel/__init__.py [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+        return {r for a in self.sequence for r in a.required_resources}
+
+    def wavelengths(self) -> Set[float]:
+        return {c.wavelength for c in self.sequence}
+
+    def constraints(self) -> Set[Constraints]:
+        return {self.constraints}
+
+
+# Since Python doesn't allow classes to self-reference, we have to make a basic group
```
Comment (stroncod) [inline common/minimodel/__init__.py:365]: I understand this is an issue with using type annotations??? Using literal string might solve that.

**Review by stroncod — APPROVED** [review]

**Comment by sraaphorst** [comment]
I'm merging this, although it is a WIP and we will need to adapt it as we go. Outstanding tasks currently include:

- Addition of the instrument configuration at the atom level.
- Decisions about where time delays fit (i.e. not in Group, but possibly in Observation) for cadences based on Bryan's feedback. Waiting for some examples to make sure that we can model cadences appropriately.

**Pull request merged**